static/js/pages/instances/FileRow.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { FC, useState } from "react";
+import { useQuery } from "@tanstack/react-query";
+import { Button, Icon } from "@canonical/react-components";
+import { queryKeys } from "util/queryKeys";
+import { fetchInstanceLogFile } from "api/instances";
+import { LxdInstance } from "types/instance";
+import DownloadButton from "pages/instances/DownloadButton";
+import { getUrlParam } from "util/helpers";
+
+interface FileRowProps {
+  instance: LxdInstance;
+  path: string;
+}
+
+const FileRow: FC<FileRowProps> = ({ instance, path }) => {
+  const fileName = path.split("/").at(-1) ?? "";
+  const fileUrl = `/ui/${instance.project}/instances/${instance.name}/logs/?file=${fileName}`;
+  const [isOpen, setOpen] = useState(getUrlParam("file") === fileName);
+
+  const {
+    data: logContent,
+    isLoading,
+    isSuccess,
+  } = useQuery({
+    queryKey: [queryKeys.instances, instance.name, queryKeys.logs, fileName],
+    queryFn: () =>
+      fetchInstanceLogFile(instance.name, instance.project, fileName),
+    enabled: isOpen,
+  });
+
+  const toggleVisibility = () => {
+    setOpen((prevState) => !prevState);
+  };
+
+  return (
+    <div className="p-code-snippet file-row">
+      <div className="p-code-snippet__header file-row-header">
+        <Button
+          appearance="base"
+          hasIcon
+          className="u-no-margin--bottom file-row-toggle"
+          aria-label={`Toggle ${fileName} visibility`}
+          aria-controls={fileName}
+          aria-hidden={!isOpen}
+          onClick={toggleVisibility}
+        >
+          <Icon name={isOpen ? "chevron-up" : "chevron-down"} />
+          <h3 className="p-code-snippet__title file-row-title">{fileName}</h3>
+        </Button>
+        <a
+          href={fileUrl}
+          aria-label={`File ${fileName} URL`}
+          className="p-button--base has-icon is-dense u-no-margin--bottom"
+        >
+          <Icon name="get-link" />
+        </a>
+        <DownloadButton
+          fileName={fileName}
+          instance={instance}
+          content={logContent ?? ""}
+        />
```

Comment:
I didn't want to use the onClick, because then we can't copy the link or open it in a new tab easily.
But removing the two helper classes from the link adjust the layout to be the same.